src/client/actions/team.ts

```diff
@@ -0,0 +1,84 @@
+import { Pokemon } from '../reducers/team'
+
+export interface UpdateTeamAction {
+    type: 'TEAM_UPDATE'
+    pokemon: Pokemon[]
+}
+
+export const handleUpdateTeam = (rawPacket: string): UpdateTeamAction => {
+    const teamData: string[] = rawPacket.split('\r\n')
+    const newPokemon: Pokemon[] = []
+    for(let i = 0; i < teamData.length; i ++) {
+        if(teamData[i] != '') {
+            const data: string[] = teamData[i].split('|')
+            
+            let currentPokemon: Pokemon
```

Comment:
I personally prefer to assign all values at once, so it's clear Pokemon is a static object